packages/office-ui-fabric-react/src/components/Icon/Icon.types.ts

```diff
@@ -70,6 +70,12 @@ export interface IIconProps extends IBaseProps, React.HTMLAttributes<HTMLElement
    */
   styles?: IStyleFunctionOrObject<IIconStyleProps, IIconStyles>;
   theme?: ITheme;
+
+  /**
+   * Flag to decide whether to cache null icon or not. If the flag is set to true, the icon is recomputed every time till
```

Comment:
I think not caching undefined results should be the default behavior (instead of adding a flag).

Reply:
I have introduced a flag just to ensure backward compatibility, just in case any component using cached icons might want to keep it as the first value stored and don't want to update the value.

Reply:
Please let me know if keeping backward compatibility is not required in this case as the existing behavior is not correct, then I will remove the flag.

Reply:
We always want the 'exception' to be the boolean prop. So if you expect most people will want to `doNotCacheNullIcon`, then we should make `CacheNullIcons` the prop, so that others can opt back into that behavior.

On the other note, backwards compatibility for us means not breaking existing prop contracts. If we make a functional change that improves the control (like not caching null icons) we don't necessarily have to support the less efficient approach as well. If anything, I'd rather leave this prop out unless someone specifically asks for it to be added back in. You can always add props later. You can never take them away :)

Reply:
Agreed with Micah. Since the new behavior should generally be considered an improvement, we shouldn't add a flag, to avoid API bloat. Then if someone files an issue with a compelling case for caching null icons (which I doubt will happen), we can consider whether to add a flag at that point.

Reply:
I have removed the prop from Icon